src/transformers/trainer.py

```diff
@@ -3571,25 +3615,26 @@ def push_to_hub(self, commit_message: Optional[str] = "End of training", blockin
         if not self.is_world_process_zero():
             return
 
-        # Cancel any async push in progress if blocking=True. The commits will all be pushed together.
-        if blocking and self.push_in_progress is not None and not self.push_in_progress.is_done:
-            self.push_in_progress._process.kill()
-            self.push_in_progress = None
+        self.create_model_card(model_name=model_name, **kwargs)
 
-        git_head_commit_url = self.repo.push_to_hub(
-            commit_message=commit_message, blocking=blocking, auto_lfs_prune=True
+        # Cancel any async push in progress and wait for the current upload to be finished.
+        if self.push_in_progress is not None and not self.push_in_progress.is_done():
+            self.push_in_progress.cancel()
```

Comment:
Same here, while with the previous workflow the final push would have pushed all the commits eventually, here we'd be cancelling one push and therefore skipping one commit